### DISCUSSION TITLE
Adventure/MiniMessage Support

### DISCUSSION THREAD

**Discussion by NickNackGus** [PlaceholderAPI/PlaceholderAPI#742] [(3) Ideas]
### Confirmation

- [X] I checked the Issues and Pull request tab for any existing issues or PRs.
- [X] My Feature Request is for **PlaceholderAPI** and not any expansion or other plugin.

### Type

New API feature

### Description

I would like to see Adventure/MiniMessage placeholder support for Placeholder API.

Mojang is moving away from legacy section symbol formatting and focusing on raw json text instead. Paper and other servers are now offering Adventure as a replacement, and since editing json messages isn't for everyone, MiniMessage is becoming a new standard for messages.

While MiniMessage has a placeholder system, it has no API for plugins to exchange placeholders. I feel PlaceholderAPI is well positioned to make transitioning away from legacy formatting and towards Adventure's implementation of raw json text easier for server owners, with older plugins' placeholders being converted by Placeholder API when using the newer format.

https://docs.adventure.kyori.net/minimessage.html#placeholder

**Comment by Andre601** [comment]
This is planned for v3

**Reply by KingsMMA** [reply]
Would I be right in assuming this is no longer planned?

**Reply by KingsMMA** [reply]
I'm happy to implement this and create a PR, is that something that would be merged?

**Reply by Andre601** [reply]
I can't say for sure whether this is still planned or not.
What I can say is, that there is a plugin that utilizes MiniMessage's custom tag functionality to allow placeholders, and it offers a papi one too.

One question right now is, what need is there for PAPI actually to support MiniMessage?
The system as of right now works fine with Strings and all you need is a plugin that parses the final String.
If you want a `String -> Component` system in PAPI then that would result in either a breaking API change, or a separate method plugins and expansions would first need to adabt...

It's a complicated mess to say the least.
